draft-ietf-jsonpath-base.md

```diff
@@ -156,77 +156,81 @@ their UTF-8 encoding.
-referred to as its root node.
-
-JSON value:
+Value:
 : As per {{-json}}, a structure complying to the generic data model of JSON, i.e.,
   composed of components such as containers, namely JSON objects and arrays, and
   atomic data, namely null, true, false, numbers, and text strings.
 
-Node:
-: A JSON value, with an emphasis on its location within the
-  Argument.  I.e., a JSON value that is identical to or contained within the JSON
-  value to which the query is applied.  A node can be viewed as a
-  combination of a (1) JSON value and (2) its location in the
-  Argument; the latter can, if desired, be represented as an Output Path.
-
-Object:
-: A JSON object as defined in {{-json}}.
-  Never used in its generic sense, e.g., for programming language objects.
+String:
+: A sequence of UTF-8 characters surrounded by either single quotes (`)
+  or double quotes (").
+  Escaping quote characters is required only for the specific character
+  which surrounds the string.
+  For example, a double quote character (") within a string surrounded
+  by single quotes (') does not need to be escaped.
+  Single quotes are preferred.
 
 Member:
-: A name/value pair in a JSON object.  (Not itself a JSON value.)
+: A name/value pair in an object.  (Not itself a value.)
 
 Name:
 : The name in a name/value pair constituting a member.  (Also known as
   "key", "tag", or "label".)
-
-Array:
-: A JSON array as defined in {{-json}}.
-  Never used in its generic sense, e.g., for programming language objects.
+  This is also used in {{-json}}, but that specification does not 
+  formally define it.
+  It is included here for completeness.
 
 Element:
-: A JSON value in an array.  (Also used with a distinct meaning in XML
+: A value in an array.  (Also used with a distinct meaning in XML
   context for XML elements.)
 
 Index:
 : A non-negative integer that identifies a specific element in an array.
-  The term, in particular in the form indexing, is also sometimes used
-  for either an index or a member name, when both arrays and objects
-  can be fed to the indexing operation.
 
 Query:
 : Short name for JSONPath expression.
 
 Argument:
-: Short name for the JSON value a JSONPath expression is applied to.
+: Short name for the value a JSONPath expression is applied to.
+
+Node:
+: The pair of a value along with its location within the argument.
+
+Root Node:
+: The unique node whose value is the entire argument.
 
 Nodelist:
-: Output of applying a query to an argument: a list of nodes within
-  that argument.
+: A list of nodes.  <!-- ordered list?  Maybe TBD by issues #27 and #60 -->
+  The output of applying a query to an argument is manifested as a list of nodes.
   While this list can be represented in JSON, e.g. as an array, the
-  nodelist is an abstract concept unrelated to JSON values.
+  Nodelist is an abstract concept unrelated to JSON values.
 
-Output Path:
+Normalized Path:
 : A simple form of JSONPath expression that identifies a node by
```

Comment:
This permits multiple possibilities. Should it be tightened up to say which types of selector may and may not be used?